app/controllers/course/forum/posts_controller.rb

```diff
@@ -51,7 +51,14 @@ def destroy
       render json: { isTopicDeleted: true }
     elsif @post.destroy
       @topic.update_column(:latest_post_at, @topic.posts.last&.created_at || @topic.created_at)
-      render json: { topicId: @topic.id, postTreeIds: @topic.posts.ordered_topologically.sorted_ids }
+      if @topic.topic_type == 'question'
+        @topic.specific.update_resolve_status
+        render json: { topicId: @topic.id,
+                       postTreeIds: @topic.posts.ordered_topologically.sorted_ids,
+                       isTopicResolved: @topic.reload.resolved? }
+      else
+        render json: { topicId: @topic.id, postTreeIds: @topic.posts.ordered_topologically.sorted_ids }
+      end
```

Comment:
Your code is duplicating the render code in line 56 and line 60.  
You should do it this way:

if topic is question
`-> update resolve status
render code

Then you only have 1 render code.

Also, you only need to update resolve status iff topic is question AND post is marked as answer.

Reply:
Roger that, and done.